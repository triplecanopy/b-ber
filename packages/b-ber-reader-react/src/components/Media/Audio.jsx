import React from 'react'
import Media from './Media'
import AudioElement from './AudioElement'

const Audio = props => (
  <Media mediaType="audio" MediaComponent={AudioElement} {...props} />
)

export default Audio
