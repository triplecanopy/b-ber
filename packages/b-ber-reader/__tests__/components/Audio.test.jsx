import React from 'react'
import renderer from 'react-test-renderer'
import Audio from '../../src/components/Media/Audio'

jest.mock(
  '../../src/lib/with-node-position',
  () => WrappedComponent => props => <WrappedComponent {...props} />
)

describe('Audio', () => {
  // Clean up logging from HOC during tests
  beforeEach(() => {
    console.warn = jest.fn()
    console.error = jest.fn()
  })

  afterEach(() => jest.clearAllMocks())

  test('renders the component', () => {
    let props
    let tree

    const ref = React.createRef()

    props = { id: 'foo', 'data-autoplay': true, controls: true }
    tree = renderer.create(<Audio elemRef={ref} {...props} />).toJSON()

    expect(tree).toMatchSnapshot()

    props = { id: 'foo', 'data-autoplay': false, controls: false }
    tree = renderer.create(<Audio elemRef={ref} {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
