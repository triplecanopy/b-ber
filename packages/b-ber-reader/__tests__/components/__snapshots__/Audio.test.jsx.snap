// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio renders the component 1`] = `
<audio
  controls={true}
  currentSpreadIndex={null}
  data-autoplay={true}
  id="foo"
  onEnded={[Function]}
  onLoadedMetadata={[Function]}
/>
`;

exports[`Audio renders the component 2`] = `
<audio
  controls={false}
  currentSpreadIndex={null}
  data-autoplay={false}
  id="foo"
  onEnded={[Function]}
  onLoadedMetadata={[Function]}
/>
`;
